Frameworks/CoreText/CTRun.mm

```diff
@@ -79,101 +88,122 @@ CTRunStatus CTRunGetStatus(CTRunRef run) {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 const CGGlyph* CTRunGetGlyphsPtr(CTRunRef run) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    _CTRun* runPtr = static_cast<_CTRun*>(run);
+    return runPtr && runPtr->_glyphs.size() ? runPtr->_glyphs.data() : NULL;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 void CTRunGetGlyphs(CTRunRef run, CFRange range, CGGlyph buffer[]) {
-    UNIMPLEMENTED();
+    _CTRun* runPtr = static_cast<_CTRun*>(run);
+    if (runPtr) {
+        if (range.location < 0) {
+            range.location = 0;
+        }
+
+        if (range.length == 0 || range.length + range.location >= runPtr->_glyphs.size()) {
+            range.length = runPtr->_glyphs.size() - range.location;
+        }
+
+        memcpy(buffer, &(runPtr->_glyphs[range.location]), sizeof(CGGlyph) * range.length);
+    }
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 const CGPoint* CTRunGetPositionsPtr(CTRunRef run) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    _CTRun* runPtr = static_cast<_CTRun*>(run);
+    return runPtr && runPtr->_glyphOrigins.size() ? runPtr->_glyphOrigins.data() : NULL;
 }
 
 /**
- @Status Caveat
- @Notes Only the starting position of the glyph run is handled.
+ @Status Interoperable
+ @Notes
 */
 void CTRunGetPositions(CTRunRef run, CFRange runRange, CGPoint* outPositions) {
```

Comment:
I guess you are OK here. There is no return status we can provide so to blow up here is fine

---
In reply to: [78227492](https://github.com/Microsoft/WinObjC/pull/989#discussion_r78227492) [](ancestors = 78227492)